### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by YoriB**
Task 4 attempted. Struggled with testing with the objectContaining jest metho and so reverted to using the toHaveProperty as it was something I am familiar with. Though I will be happy if shown otherwise. Did not do extra error handling as I am not sure if it is an overkill at this stage.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -14,33 +14,44 @@ afterAll(() => {
 });
 
 describe('app', () => {
-  describe('GET /api/topics', () => {
-    test('should return a status :200 ',() => {
+  describe.only('GET /api/topics', () => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:17]: remove `.only` before committing your code

File: __tests__/app.test.js [hunk 2]
```diff
@@ -14,33 +14,44 @@ afterAll(() => {
...
-          expect(body.length).toEqual(3);
-        });
-      });
-    })
-
-      test('return a body containing topics for slug and description', () => {
-            return request(app)
-            .get('/api/topics')
-            .then(({body}) => {      
+          expect(body.length).toEqual(3)
```
Comment (alexdevdavis) [inline __tests__/app.test.js:22]: nicely done

File: __tests__/app.test.js [hunk 3]
```diff
@@ -14,33 +14,44 @@ afterAll(() => {
...
+          .expect(200)
+          .then(({body}) => {
+            expect(true).toEqual(true);        
+          });
+        });      
+  test('return a body containing an array of objects with following properties', () => {
+        return request(app)
+        .get('/api/articles')
+        .then(({body}) => { 
+          console.log(body)
```
Comment (alexdevdavis) [inline __tests__/app.test.js:44]: can remove this floaty log

File: __tests__/app.test.js [hunk 4]
```diff
@@ -14,33 +14,44 @@ afterAll(() => {
...
+      body.forEach(article => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:45]: remember to do a length check before a forEach

File: __tests__/app.test.js [hunk 5]
```diff
@@ -14,33 +14,44 @@ afterAll(() => {
...
+          expect(article).toHaveProperty('author')
+          expect(article).toHaveProperty('title')  
+          expect(article).toHaveProperty('article_id') 
+          expect(article).toHaveProperty('topic') 
+          expect(article).toHaveProperty('created_at')  
+          expect(article).toHaveProperty('votes') 
+          expect(article).toHaveProperty('article_img_url')     
+      })
```
Comment (alexdevdavis) [inline __tests__/app.test.js:53]: this needs to check for a comment_count property too

File: models/model.js [hunk 6]
```diff
@@ -3,48 +3,15 @@ const { db } = require('../db/connection');
...
-    });
+  return db.query(`SELECT * FROM topics`).then((result) => {
+    return result.rows;
+  });
 };
 
+const fetchArticles = () => {
+  return db.query(`SELECT * FROM articles ORDER BY rating DESC`).then((result) => {
+    return result.rows;
+  });
```
Comment (alexdevdavis) [inline models/model.js:14]: use the aggregate function COUNT() and AS comment_count to add the desired property that ticket 4 is asks for

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
Before going on to ticket five, ensure ticket 4 includes `comment_count` property

Otherwise, code is looking good 👍

File: __tests__/app.test.js (hunk 1 above)
Comment (YoriB) [reply __tests__/app.test.js:17]: OK will implement all your suggestions. Thanks for your feedback.

**Pull request merged**